### ISSUE TITLE
Add ignoreAtRules: [] option to max-nesting-depth

### ISSUE THREAD

**Issue by aaronbushnell** [stylelint/stylelint#1705]
I'd like to ignore nested media queries when using `max-nesting-depth`.

Given:
```json
{
  "rules": {
    "max-nesting-depth": [1, {
      ignore: ["at-rules-without-declaration-blocks"]
    }],
  }
}
```

I expect:
```css
.module {
  padding: 1rem;

  .a-nested-item {
    color: red;

    /* An error should occur here*/
    .a-further-nested-item {
      color: green;
    }

    /* This should not throw an error */
    @media (min-width: 50em) {
      color: green;
    }
  }
}
```

What I got:
```css
.module {
  padding: 1rem;

  .a-nested-item {
    color: red;

    /* An error occurred here*/
    .a-further-nested-item {
      color: green;
    }

    /* Another error occurred here*/
    @media (min-width: 50em) {
      color: green;
    }
  }
}
```

- **Stylelint version:** 7.0.3
- **Running via:** PostCSS + Atom plugin
- **Related to non-standard syntax?**: Yes, nesting.

**Comment by jeddy3** [comment]
@stylelint/core Any objections to adding an `ignoreAtRules: []` secondary option to this rule? If not, I can point @aaronbushnell in the direction of how to add it.

**Comment by ntwb** [comment]
SGTM

**Comment by jeddy3** [comment]
It looks like no one is raising any objections to a new secondary option. So, I'll label this up as "help wanted".

@aaronbushnell Would you like to contribute this option? We recently added an `ignoreAtRules: []` secondary option to [`at-rule-no-unknown`](https://github.com/stylelint/stylelint/blob/master/src/rules/at-rule-no-unknown). You'll find ideas for everything you need there: [code](https://github.com/stylelint/stylelint/blob/master/src/rules/at-rule-no-unknown/index.js#L33), [tests](https://github.com/stylelint/stylelint/blob/master/src/rules/at-rule-no-unknown/__tests__/index.js#L73-L108) and [docs](https://github.com/stylelint/stylelint/tree/master/src/rules/at-rule-no-unknown#ignoreatrules-regex-string). You'll need to adapt them all so the behaviour is about ignoring the at-rule in the depth calculation, rather than ignoring the at-rule outright.

Step-by-step instructions for adding a new option are in the [developer guide](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#adding-options-to-existing-rules), along with how to [run tests](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#running-tests).

**Comment by Nirazul** [comment]
That's a feature I could definitely use!

**Comment by aaronbushnell** [comment]
Thanks, @jeddy3! I'm a tad tied up, but I'd love to take a peek at some point. Thanks for the helpful information here!

**Comment by gaidarenko** [comment]
Let me add this option.

**Comment by jeddy3** [comment]
> Let me add this option.

Sgtm.

**Comment by aaronbushnell** [comment]
Thanks so much, @gaidarenko and @jeddy3!
